apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/frontend/GraphQLParser.kt

```diff
@@ -76,12 +73,12 @@ object GraphQLParser {
    * A specialized version that works on multiple files and can also inject fragments from another
    * compilation unit
    */
-  fun parseExecutableFiles(files: Set<File>, schema: Schema, injectedFragmentDefinitions: List<GQLFragmentDefinition>): ParseResult<List<GQLDocument>> {
+  fun parseExecutableFiles(files: Set<File>, schema: Schema, injectedFragmentDefinitions: List<GQLFragmentDefinition>): Pair<List<GQLDocument>, List<Issue>> {
```

Comment:
why `ParseResult ` can't be used here?

Reply:
Since `parseExecutableFiles` returns a list, it will never be null. It'll be empty instead, which is conveyed by the `Pair` type. This also signals that this API behaves inconsistently from the others in the file. i.e. it does more things like checking duplicates and non-executable definitions.

This might mean that we should move this code somewhere else and keep this file 100% about the spec.